### PR TITLE
Set gasPrice and gasLimit dynamically

### PR DESCRIPTION
**Pull request by marcosio**
To make the use of the web3j-eea library more flexible to more use cases, we propose to set the gasPrice or the gasLimit globally or in each transaction.

We propose to change EEAClient constructor to receive gasPrice and gasLimit as optional parameters, with the default values of the current GAS_PRICE and GAS_LIMIT constants. In this way, users will be able to assign global default values to the gasPrice and gasLimit, which will be applied to all transactions.

On the other hand, if the user sets gasPrice and gasLimit when issuing a transaction (by setting this values in the "options" argument of the sendRawTransaction function), they will be applied only to that transaction, overwriting the global default values mentioned in the previous paragraph.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -121,6 +118,7 @@ function EEAClient(web3, chainId) {
    * - options.nonce
    * - options.to
    * - options.data
```
Comment (achraf17) [inline src/index.js:120]: add the `gasPrice` and `gasLimit` to the JSDoc and their defaults value

**Comment by achraf17** [comment]
+1

File: src/index.js (hunk 1 above)
Comment (marcosio) [reply src/index.js:122]: Done, please review it.

**Comment by achraf17** [comment]
LGTM

**Review by macfarla — APPROVED** [review]
LGTM. one minor comment

File: example/multiNodeExample/deployContract.js [hunk 2]
```diff
@@ -10,14 +10,15 @@ const binary = fs.readFileSync(
...
+const web3 = new EEAClient(new Web3(besu.node1.url), 2018, 0, 2999999);
 
 const createPrivateEmitterContract = () => {
   const contractOptions = {
     data: `0x${binary}`,
     privateFrom: orion.node1.publicKey,
     privateFor: [orion.node2.publicKey],
-    privateKey: besu.node1.privateKey
+    privateKey: besu.node1.privateKey,
+    gasLimit: 3000000
```
[the remark this answers is not in the document]
Comment (marcosio) [reply example/multiNodeExample/deployContract.js:21]: Yes, this change was included to illustrate its use and that it was in a test.

**Pull request merged**